drivers/ufs/ufs.c

```diff
@@ -705,11 +705,27 @@ static void ufs_enum(void)
 	}
 }
 
+static void ufs_get_device_info(struct ufs_dev_desc *card_data)
+{
+	uint8_t desc_buf[DESC_DEVICE_MAX_SIZE];
+
+	ufs_query(QUERY_READ_DESC, DESC_TYPE_DEVICE, 0, 0,
+				(uintptr_t)desc_buf, DESC_DEVICE_MAX_SIZE);
+
+	/*
+	 * getting vendor (manufacturerID) and Bank Index in big endian
+	 * format
+	 */
+	card_data->wmanufacturerid = desc_buf[DEVICE_DESC_PARAM_MANF_ID] << 8 |
```

Comment:
Static analysis errors:
- Ideally wrap desc_buf[DEVICE_DESC_PARAM_MANF_ID] << 8 in parenthesis rather than rely on operator precedence.
- wmanufacturerid is uint16_t - best to explicitly cast right-hand side of assignment to uint16_t (as desc_buf[] is uint8_t).